games-puzzle/gemdropx/gemdropx-0.9-r2.ebuild

```diff
@@ -1,47 +1,43 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-inherit desktop
+EAPI=8
 
-DESCRIPTION="A puzzle game where it's your job to clear the screen of gems"
+inherit desktop toolchain-funcs
+
+DESCRIPTION="Puzzle game where it's your job to clear the screen of gems"
 HOMEPAGE="http://www.newbreedsoftware.com/gemdropx/"
-SRC_URI="ftp://ftp.sonic.net/pub/users/nbs/unix/x/gemdropx/${P}.tar.gz"
+SRC_URI="ftp://ftp.tuxpaint.org/unix/x/gemdropx/src/${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE=""
 
 RDEPEND="
-	>=media-libs/libsdl-1.2.3-r1[joystick,video]
-	>=media-libs/sdl-mixer-1.2.1[mod]
-"
+	media-libs/libsdl[sound,joystick,video]
+	media-libs/sdl-mixer[mod]"
 DEPEND="${RDEPEND}"
 
-src_prepare() {
-	default
-	sed -i \
-		-e '/^CC/d' \
-		-e '/^CXX/d' \
-		-e 's/CXX/CC/' \
-		-e 's/-o/$(LDFLAGS) -o/' \
-		Makefile || die
-
-	find data/ -type d -name .xvpics -exec rm -rf \{\} +
-}
+DOCS=( AUTHORS.txt CHANGES.txt ICON.txt README.txt TODO.txt )
 
 src_compile() {
-	emake \
-		DATA_PREFIX="/usr/share/${PN}" \
-		XTRA_FLAGS="${CFLAGS}"
+	local myemakeargs=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
```

Comment:
Thanks for the detailed explanation (also with your previous comment).
I've changed the ebuild now as following:
```
src_prepare() {
	default
	sed -i \
		-e 's/$(CFLAGS) -o/$(LDFLAGS) -o/' \
		Makefile || die
	rm -r data/images/.xvpics || die
}

src_compile() {
	local myemakeargs=(
		CXX="$(tc-getCC)"
		DATA_PREFIX="${EPREFIX}/usr/share/${PN}"
		XTRA_FLAGS="${CFLAGS}"
	)
	emake "${myemakeargs[@]}"
}
```

However i still see that `gcc` is called directly once.
```
gcc -Wall -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -DDATA_PREFIX=\"/usr/share/gemdropx\" -DSOUND -DJOYSTICK -DNOT_EMBEDDED -O2 -pipe -march=bdver2 -mprefer-avx128   -c -o gemdropx.o gemdropx.c
```

Adding `CC="$(tc-getCC)"` would fix it, but i'm actually not really sure where it even comes from.

Reply:
I didn't mean to say should remove CC=, just update CXX to point to CC as well

The $(CC) is used by implicit rules.

Reply:
Ahhh, ok, now i got it.
to recap: `$(CC) ..` is the implicit rule and thus used by the makefile for building the `.o` file (out of the `.c` file). `CXX` isn't really needed for linking (since no c++ files are present) and the previous `sed` simply removed the fixed `gcc` calls and replaced CXX by CC.
Now we don't replace `CXX` by `CC` but simply set `tc-getCC` to both. Once for the implicit `CC` variable and once for the `CXX` variable in the makefile.

I've updated the ebuild once again, and sorry for my stupidity. I really had to look a few times to understand it.

Reply:
Don't worry about it, it's the Makefile itself that's silly/misleading -- I was almost tempted to suggest dropping it entirely (easy when it's a single .c file) but, along with all the -D that need to be set, felt it didn't simplify much so I dropped the idea.